shell/components/form/UnitInput.vue

```diff
@@ -235,7 +235,7 @@ export default {
     :required="required"
     :placeholder="placeholder"
     :hide-arrows="hideArrows"
-    @change="update($event.target.value)"
+    @update:value="update"
```

Comment:
Have you tested this component with pages where it's consuming the emitter `update`?  In this modified version it seems to pass directly the whole event instead of the value, no?
Also is not changing the emitter from `update` to `change` affecting other components/views?

Reply:
Imo, the problem lies with how `UnitInput` gets the value from it's child component `LabeledInput`. From my tests, what is passed upstream is the value that `UnitInput` parses. It can be a simple integer (in `number` or `string` format) or a string with units. I think that part looks good, tbh.
From searching the code there are 68 instances of `UnitInput` being used... I haven't checked them all, but we do have tests. I haven't found any problems with a few that I looked on but let's see what the tests pick up

Reply:
Yes I also double-checked in the `LabeledInput` and there's a check on that and it's just inconsistent how values are emitted, but the emission seems safe from your side.

```ts
      const value = typeof val === 'string' ? val : (val?.target as HTMLInputElement)?.value;
```